packages/jmix-addon-data-tools/src/entity-inspector/gqlQueryGeneration.ts

```diff
@@ -0,0 +1,120 @@
+import { 
+  getPropertyInfo, 
+  isAssociation, 
+  isComposition, 
+  isRelationProperty, 
+  Metadata, 
+  MetaPropertyInfo,
+  unCapitalizeFirst,
+  findEntityMetadata
+} from "@haulmont/jmix-react-core";
+import {getAllEntityPropertiesNames} from "./EntityInspector.helpers";
+import {EntityNamesInfo} from "./EntityInspector.types";
+
+export enum GqlQueryType {
+  DataList = 'dataList',
+  DataById = 'dataById',
+  Upsert = 'upsert'
+}
+
+export function generateGqlQuery(
```

Comment:
It will be more readable, if we can separate function for blocks related each to one query generation. Also, it will be very helpful to see example of generated query in comments.